uefi-test-runner/build.py

```diff
@@ -101,27 +102,24 @@ def run_qemu(headless):
 
         # Mount the built examples directory.
         '-drive', f'format=raw,file=fat:rw:{examples_dir}',
+
+        # Connect the serial port to the host. OVMF is kind enough to connect
+        # the UEFI stdout and stdin to that port too.
+        '-serial', 'stdio',
     ]
 
-    # When running in headless mode we don't have video
+    # When running in headless mode we don't have video, but we can still have
+    # QEMU emulate a display and take screenshots from it.
+    qemu_flags.extend(['-vga', 'std'])
     if headless:
-        # Disable window
-        qemu_flags.append('-nographic')
-
-        # Redirect all output to stdio
-        qemu_flags.extend(['-serial', 'stdio'])
-    else:
-        # Use a standard VGA for graphics.
-        qemu_flags.extend(['-vga', 'std'])
+        # Do not attach a window to QEMU's display
+        qemu_flags.extend(['-display', 'none'])
```

Comment:
This option is more orthogonal than -nographic, which does other things like redirecting serial & parallel ports.